include/psa/crypto_extra.h

```diff
@@ -1901,6 +1901,342 @@ static inline struct psa_pake_operation_s psa_pake_operation_init( void )
+ *
+ */
+struct psa_sign_interruptible_context_s
+{
+    size_t MBEDTLS_PRIVATE(num_ops);
+};
+
+typedef struct psa_sign_interruptible_context_s psa_sign_interruptible_context;
+
+/**
+ * \brief                       Set the maximum number of basic operations done
+ *                              in a row.
+ *
+ * \note                        Although the interpretation of this value is
+ *                              implicitly implementation defined, this value
+ *                              needed a default, thus this value defaults to
+ *                              1, which means the absolute minimum of work
+ *                              will be done each interruptible call.
+ *
+ * \note                        If more operations are needed to complete a
+ *                              computation, #PSA_ERROR_IN_PROGRESS will be
+ *                              returned by the function performing the
+ *                              computation. It is then the caller's
+ *                              responsibility to either call again with the
+ *                              same parameters until it returns 0 or an error
+ *                              code; or to free the relevant context if the
+ *                              operation is to be aborted.
+ *
+ * \note                        It is strictly required that all input
+ *                              parameters and the restart context be the same
+ *                              on successive calls for the same operation,
+ *                              but output parameters need not be the same;
+ *                              they must not be used until the function
+ *                              finally returns 0.
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param max_ops               The maximum number of basic operations to
+ *                              be done in a row. Passing zero here has the
+ *                              same effect as passing the maximum value - i.e
+ *                              the operation will be done in one go. Lower
+ *                              (non-zero) values mean signature functions
+ *                              will block for a lesser maximum amount of time
+ *                              and conversely larger values will mean
+ *                              blocking for a larger maximum amount of time.
+ *                              The effect of changing this value will depend
+ *                              on the key used, the curve used and the
+ *                              underlying implementation, so we provide /c
+ *                              psa_sign_interruptible_get_num_ops() in order
+ *                              to help with tuning this value.
+ *
+ */
+void psa_sign_interruptible_set_max_ops( size_t max_ops );
```

Comment:
Is the intention that we allow different `max_ops` settings for each interruptible operation (just sign now, but in the future could be verify, ecdh...)? This would be a problem with the current implementation (and more importantly) API in Mbed TLS, which has a single global setting that applies to all operations.